slack-message.c

```diff
@@ -342,18 +409,34 @@ void slack_json_to_html(GString *html, SlackAccount *sa, json_value *message, Pu
 
 	if (!g_strcmp0(subtype, "me_message"))
 		g_string_append(html, "/me ");
-	else if (subtype && flags)
+	else if (flags && subtype && strcmp(subtype, "thread_broadcast") != 0)
 		*flags |= PURPLE_MESSAGE_SYSTEM;
 
-	json_value *thread = json_get_prop(message, "thread_ts");
-	if (thread) {
-		time_t tt = slack_parse_time(thread);
-		g_string_append(html, purple_time_format(localtime(&tt)));
-		g_string_append(html, "⤷  ");
+	gboolean display_parent_indicator = purple_account_get_bool(sa->account, "display_parent_indicator", TRUE);
+	const char *ts = json_get_prop_strptr(message, "ts");
+	const char *thread = json_get_prop_strptr(message, "thread_ts");
+	if (thread && (display_parent_indicator || g_strcmp0(ts, thread))) {
+		if (g_strcmp0(ts, thread))
+			g_string_append(html, "⤷ ");
+		else
+			g_string_append(html, "◈ ");
```

Comment:
These need to be configurable. Some clients (my finch included) may not be unicode-friendly.

Reply:
New configuration options added! Those unicode characters are still the defaults, but I reckon that's ok, since attachments already have a unicode default.